src/Select/Select.tsx

```diff
@@ -29,45 +30,40 @@ type ReactSelectStateManager = {
   blur: () => void;
 };
 
-export type SelectProps = {
-  options?: any[];
+// NOTE: We recreate these props as upstream doesn't export them. Note also that
+// we have a default value for windowThreshold, therefore this param is optional.
+interface WindowedSelectProps extends SelectProps {
   windowThreshold?: number;
-  filterOption?: (...args: any[]) => any;
-  autocomplete?: boolean;
-  disabled?: boolean;
-  error?: boolean;
-  errorMessage?: string;
-  errorList?: string[];
+}
+
+interface NDSOptionType {
+  label: string;
+  value: unknown;
+}
+
+interface CustomProps<Option, IsMulti extends boolean, Group extends GroupBase<Option>> {
+  autocomplete?: SelectProps<Option, IsMulti, Group>["isSearchable"];
   labelText?: string;
-  helpText?: any;
-  noOptionsMessage?: Function;
   requirementText?: string;
-  id?: string;
-  initialIsOpen?: boolean;
-  menuPosition?: string;
-  menuPlacement?: string;
+  helpText?: ReactNode;
+  disabled?: SelectProps<Option, IsMulti, Group>["isDisabled"];
+  errorMessage?: string;
+  errorList?: string[];
+  initialIsOpen?: SelectProps<Option, IsMulti, Group>["defaultMenuIsOpen"];
+  multiselect?: SelectProps<Option, IsMulti, Group>["isMulti"];
   maxHeight?: string;
-  multiselect?: boolean;
-  name?: string;
-  onBlur?: (...args: any[]) => any;
-  onChange?: (...args: any[]) => any;
-  placeholder?: string;
-  required?: boolean;
-  value?: any;
-  defaultValue?: any;
-  className?: string;
-  classNamePrefix?: string;
-  menuIsOpen?: boolean;
-  onMenuOpen?: (...args: any[]) => any;
-  onMenuClose?: (...args: any[]) => any;
-  onInputChange?: (...args: any[]) => any;
-  components?: any;
-  closeMenuOnSelect?: boolean;
-  "aria-label"?: string;
   size?: ComponentSize;
-  isClearable?: boolean;
-  [key: string]: any; // Allow for custom props to be passed and used inside custom components using the `selectProps` prop
-};
+  error?: boolean;
+  options: NDSOptionType[];
+  onChange?: (newValue: unknown) => void;
+  [key: string]: any;
```

Comment:
This is required for backwards compatibility, we currently allow passing any random prop to the select component. We have a [story that covers this](https://master--5f60c6c285eaad0022dce67f.chromatic.com/?path=/story/components-select--with-custom-props).

This is also part of the react-select API. They do recommend typing it [Custom Select props](https://react-select.com/typescript#custom-select-props), but we don't know what props someone might pass so it can't be done in NDS.

We can remove it and not allow custom props, but we will be creating a breaking change and a new version of NDS. 


what are your thoughts @jherdman?

Reply:
IMHO let's just keep it as it maps to React Select and should result in few surprises.